tests/unit/App.spec.ts

```diff
@@ -0,0 +1,152 @@
+      getViewport() {
+        return {
+          setStack: vi.fn(),
+          render: vi.fn(),
+        };
+      }
+      enableElement() {}
+    },
+    Types: vi.fn(),
+    Enums: {
+      ViewportType: {
+        STACK: "", // "stack",
+      },
+    },
+    metaData: vi.fn(),
+    init: vi.fn(),
+    getConfiguration: vi.fn().mockImplementation(() => {
+      return { rendering: "" };
+    }),
+    registerImageLoader: vi.fn(),
+    isCornerstoneInitialized: vi.fn(),
+  };
+});
+
+vi.mock("@cornerstonejs/dicom-image-loader", () => ({
+  wadouri: {
+    loadImage: (imgId) => ({
+      promise: Promise.resolve({}),
+    }),
+  },
+  external: vi.fn(),
+  configure: vi.fn(),
+  loadImage: vi.fn(),
+  webWorkerManager: {
+    initialize: vi.fn().mockImplementation(() => {}),
+  },
+}));
+
+vi.mock("dicom-parser", () => ({
+  default: {},
+}));
+
+describe("App component", () => {
+  describe("Methods", () => {
+    vi.spyOn(App.methods, "fetchVipMetadataInformation").mockImplementation(
+      vi.fn()
+    );
+    vi.spyOn(App.methods, "fetchMetadataInformation").mockImplementation(
+      vi.fn()
+    );
+    const wrapper = getWrapper();
+
+    describe("method: wadouri", () => {
+      it('should add "wadouri" prefix', async () => {
+        expect(await wrapper.vm.addWadouriPrefix("https://dummy_url")).toBe(
+          "wadouri:https://dummy_url"
+        );
+      });
+    });
+
+    describe("method: formatTime", () => {
+      const invalidTimeFormats = [undefined, "155"];
+      it("should format a valid time", async () => {
+        expect(await wrapper.vm.formatTime("155614.6300")).toBe("15:56:14");
+      });
+      it.each(invalidTimeFormats)(
+        "should return undefined for invalid time",
+        (invalidtime) => {
+          expect(wrapper.vm.formatTime(invalidtime)).toBe(undefined);
+        }
+      );
+    });
+
+    describe("method: formatDate", () => {
+      const invalidDateformats = [undefined, "1901010"];
+      it("should format a valid date", () => {
+        expect(wrapper.vm.formatDate("19010101")).toBe("Jan 1, 1901");
+      });
+      it.each(invalidDateformats)(
+        "should return undefined for invalid date",
+        (invalidDate) => {
+          expect(wrapper.vm.formatDate(invalidDate)).toBe(undefined);
+        }
+      );
+    });
+
+    describe("method: formatDateAndTime", () => {
+      const invalidDateTimeCombinations = [
+        {
+          date: undefined,
+          time: undefined,
+        },
+        {
+          date: "1996082",
+          time: "093801",
+        },
+        {
+          date: "19960823",
+          time: "09380",
+        },
+        {
+          date: "1996082",
+          time: "09380",
+        },
+      ];
+      it("should format a valid date", () => {
+        expect(wrapper.vm.formatDateAndTime("19960823", "093801")).toBe(
+          "Aug 23, 1996, 9:38 AM"
+        );
+      });
+      it.each(invalidDateTimeCombinations)(
```

Comment:
no need for extra variable if not reused
```suggestion
      it.each([
        {
          date: undefined,
          time: undefined,
        },
        {
          date: "1996082",
          time: "093801",
        },
        {
          date: "19960823",
          time: "09380",
        },
        {
          date: "1996082",
          time: "09380",
        },
      ])(
```